src/test/java/io/xstefank/wildfly/bot/PRCommitsQuantityCheckTest.java

```diff
@@ -0,0 +1,74 @@
+package io.xstefank.wildfly.bot;
+
+import io.quarkiverse.githubapp.testing.GitHubAppTest;
+import io.quarkiverse.githubapp.testing.GitHubAppTesting;
+import io.quarkus.test.junit.QuarkusTest;
+import io.xstefank.wildlfy.bot.config.CommitsQuantity;
+import io.xstefank.wildlfy.bot.format.checks.CommitsQuantityCheck;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHCommitState;
+import org.kohsuke.github.GHEvent;
+import org.kohsuke.github.GHRepository;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@QuarkusTest
+@GitHubAppTest
+public class PRCommitsQuantityCheckTest {
+
+    private String wildflyConfigFile;
+
+    @BeforeEach
+    void SetUp() {
+        wildflyConfigFile =
+                "wildfly:\n" +
+                        "  rules:\n" +
+                        "    - title: \"test\"\n" +
+                        "    - body: \"test\"\n" +
+                        "      notify: [xstefank,petrberan]\n"+
+                        "  format:\n" +
+                        "    commits-quantity:\n" +
+                        "      quantity: \"1-2\"\n" +
+                        "      message: \"Too many commits in PR!\"";
+    }
+
+    @Test
+    void wildflyConfigFileQuantityIsNullTest() {
+        CommitsQuantity commitsQuantity = new CommitsQuantity();
+
+        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
+                () -> new CommitsQuantityCheck(commitsQuantity));
+        Assertions.assertEquals("Quantity was not set", thrown.getMessage());
+    }
+
+    @Test
+    void commitsQuantityCheckFailMatchTest() throws IOException {
+        GitHubAppTesting.given()
+                .github(mocks -> mocks.configFileFromString("wildfly-bot.yml", wildflyConfigFile))
+                .when().payloadFromClasspath("/pr-fail-checks.json")
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    GHRepository repo = mocks.repository("tutorial-quarkus-github-app");
+                    Mockito.verify(repo).createCommitStatus("3de181159483ae3f2103e258cd2a6e7f9bafb3b8",
+                            GHCommitState.ERROR, "", "\u274C commits-quantity: Too many commits in PR!", "Format");
+                });
+    }
+
+    @Test
+    void commitsQuantityCheckSuccessMatchTest() throws IOException {
+        GitHubAppTesting.given()
+                .github(mocks -> mocks.configFileFromString("wildfly-bot.yml", wildflyConfigFile))
+                .when().payloadFromClasspath("/pr-success-checks.json")
```

Comment:
this should probaly have more descriptive name, might be confusing what pr-succes/failure means when you have more tests